src/inference/include/openvino/runtime/infer_request.hpp

```diff
@@ -102,28 +102,72 @@ class OPENVINO_RUNTIME_API InferRequest {
      */
     void set_tensor(const ov::Output<ov::Node>& port, const Tensor& tensor);
 
+    /**
+     * @brief Sets batch of tensors for input data to infer by input name
+     * Model input shall have batch dimension and number of 'tensors' shall match with batch size
+     * Current version supports set tensors to model inputs only. In case if `name` is associated
+     * with output (or any other non-input node) - exception will be thrown
+     *
+     * @param name Name of input tensor.
+     * @param tensors Input tensors for batched infer request. The type of each tensor must match the model
+     * input precision and size (except batch dimension). Total size of tensors shall match with input's size
+     */
+    void set_tensors(const std::string& name, const std::vector<Tensor>& tensors);
+
+    /**
+     * @brief Sets batch of tensors for input data to infer by input name
+     * Model input shall have batch dimension and number of 'tensors' shall match with batch size
+     * Current version supports set tensors to model inputs only. In case if `name` is associated
+     * with output (or any other non-input node) - exception will be thrown
+     *
+     * @param port Port of input tensor.
+     * @param tensors Input tensors for batched infer request. The type of each tensor must match the model
+     * input precision and size (except batch dimension). Total size of tensors shall match with input's size
+     */
+    void set_tensors(const ov::Output<const ov::Node>& port, const std::vector<Tensor>& tensors);
```

Comment:
In the conversation above, my question was: What if user uses this method for outputs by mistake? Maybe set_input_tensors? Because these methods are applicable only for inputs.

Reply:
We actually are going to add 'set_output_tensors' API in next PRs. So this 'set_tensors' will work for both inputs and outputs